### PR TITLE
refactor with cmake

### PR DESCRIPTION
**Pull request by Smityz**
This PR refactored the project with CMake, which brings the following benefits:
1. Makes it easier for other C++ projects to integrate
2. Supports mainstream IDEs
3. No need to install cxxbridge during the project build and hide the code generated by cxxbridge.

### CODE REVIEW

File: example/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+project(Example)
+cmake_minimum_required(VERSION 3.5)
+set (CMAKE_CXX_STANDARD 17)
+
+# using remote repository
+# include(ExternalProject)
+# set(EXTERNAL_INSTALL_LOCATION ${CMAKE_BINARY_DIR}/external)
+# ExternalProject_Add(tikv
+#     GIT_REPOSITORY https://github.com/Smityz/client-cpp.git
```
Comment (Smityz) [inline example/CMakeLists.txt:9]: This URL is for easier testing now, will be modified after approving

**Comment by Smityz** [comment]
@andylokandy @disksing Hi masters, PTAL

**Comment by andylokandy** [comment]
Would you be interested in adding a github action workflow for this newly built facility?

**Comment by Smityz** [comment]
> Would you be interested in adding a github action workflow for this newly built facility?

Done, you can check CI here https://github.com/Smityz/client-cpp/actions/runs/5701746980/job/15452747821

**Comment by andylokandy** [comment]
Thank you!

**Pull request merged**